xcp_d/utils/hcp2fmriprep.py

```diff
@@ -11,158 +11,287 @@
+            sub_ent=subject_id,
+        )
+
+    return participant_ids
+
+
+def convert_hcp_to_bids_single_subject(in_dir, out_dir, sub_ent):
+    """Convert HCP derivatives to BIDS-compliant derivatives for a single subject.
+
+    Parameters
+    ----------
+    in_dir : str
+        Path to the subject's HCP derivatives.
+    out_dir : str
+        Path to the output fMRIPrep-style derivatives folder.
+    sub_ent : str
+        Subject identifier, with "sub-" prefix.
+    """
+    assert isinstance(in_dir, str)
+    assert os.path.isdir(in_dir)
+    assert isinstance(out_dir, str)
+    assert isinstance(sub_ent, str)
+
+    sub_id = sub_ent.replace("sub-", "")
+
+    volspace = "MNI152NLin6Asym"
+    volspace_ent = f"space-{volspace}"
+    res_ent = "res-2"
+
+    anat_dir_orig = os.path.join(in_dir, "MNINonLinear")
+
+    subject_dir_fmriprep = os.path.join(out_dir, sub_ent)
+    anat_dir_fmriprep = os.path.join(subject_dir_fmriprep, "anat")
+    func_dir_fmriprep = os.path.join(subject_dir_fmriprep, "func")
+    os.makedirs(anat_dir_fmriprep, exist_ok=True)
+    os.makedirs(func_dir_fmriprep, exist_ok=True)
+
+    # Get necessary files
+    csf_mask = pkgrf("xcp_d", f"/data/masks/{volspace_ent}_{res_ent}_label-CSF_mask.nii.gz")
+    wm_mask = pkgrf("xcp_d", f"/data/masks/{volspace_ent}_{res_ent}_label-WM_mask.nii.gz")
+
+    # A dictionary of mappings from HCP derivatives to fMRIPrep derivatives.
+    # Values will be lists, to allow one-to-many mappings.
+    copy_dictionary = {}
+
+    # Collect anatomical files to copy
+    t1w_orig = os.path.join(anat_dir_orig, "T1w_restore.nii.gz")
+    t1w_fmriprep = os.path.join(
+        anat_dir_fmriprep,
+        f"{sub_ent}_{volspace_ent}_{res_ent}_desc-preproc_T1w.nii.gz",
+    )
+    copy_dictionary[t1w_orig] = [t1w_fmriprep]
+
+    brainmask_orig = os.path.join(anat_dir_orig, "brainmask_fs.nii.gz")
+    brainmask_fmriprep = os.path.join(
+        anat_dir_fmriprep,
+        f"{sub_ent}_{volspace_ent}_{res_ent}_desc-brain_mask.nii.gz",
+    )
+    copy_dictionary[brainmask_orig] = [brainmask_fmriprep]
+
+    # NOTE: What is this file for?
+    ribbon_orig = os.path.join(anat_dir_orig, "ribbon.nii.gz")
+    ribbon_fmriprep = os.path.join(
+        anat_dir_fmriprep,
+        f"{sub_ent}_{volspace_ent}_{res_ent}_desc-ribbon_T1w.nii.gz",
+    )
+    copy_dictionary[ribbon_orig] = [ribbon_fmriprep]
+
+    dseg_orig = os.path.join(anat_dir_orig, "aparc.a2009s+aseg.nii.gz")
+    dseg_fmriprep = os.path.join(
+        anat_dir_fmriprep,
+        f"{sub_ent}_{volspace_ent}_{res_ent}_desc-aparcaseg_dseg.nii.gz",
+    )
+    copy_dictionary[dseg_orig] = [dseg_fmriprep]
+
+    # Grab transforms
+    t1w_to_template_orig = pkgrf("xcp_d", "/data/transform/itkIdentityTranform.txt")
+    t1w_to_template_fmriprep = os.path.join(
+        anat_dir_fmriprep,
+        f"{sub_ent}_from-T1w_to-{volspace}_mode-image_xfm.txt",
+    )
+    copy_dictionary[t1w_to_template_orig] = [t1w_to_template_fmriprep]
```

Comment:
Since the input file is the same, this will overwrite its entry in the copy_dictionary.
```suggestion
    identity_xfm = pkgrf("xcp_d", "/data/transform/itkIdentityTranform.txt")
    t1w_to_template_fmriprep = os.path.join(
        anat_dir_fmriprep,
        f"{sub_ent}_from-T1w_to-{volspace}_mode-image_xfm.txt",
    )
    copy_dictionary[identity_xfm] = [t1w_to_template_fmriprep]
```